downloader/src/main/kotlin/JavaBootstrapper.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2024 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.downloader
+
+import eu.hansolo.jdktools.ArchiveType
+import eu.hansolo.jdktools.Latest
+import eu.hansolo.jdktools.LibCType
+import eu.hansolo.jdktools.Match
+import eu.hansolo.jdktools.OperatingSystem
+import eu.hansolo.jdktools.PackageType
+import eu.hansolo.jdktools.TermOfSupport
+import eu.hansolo.jdktools.util.Helper
+
+import io.foojay.api.discoclient.DiscoClient
+import io.foojay.api.discoclient.pkg.Scope
+
+import java.io.File
+
+import kotlin.time.measureTime
+import kotlin.time.measureTimedValue
+
+import org.apache.logging.log4j.kotlin.logger
+
+import org.ossreviewtoolkit.utils.common.safeMkdirs
+import org.ossreviewtoolkit.utils.common.unpack
+import org.ossreviewtoolkit.utils.ort.Environment
+import org.ossreviewtoolkit.utils.ort.downloadFile
+import org.ossreviewtoolkit.utils.ort.okHttpClient
+import org.ossreviewtoolkit.utils.ort.ortToolsDirectory
+
+object JavaBootstrapper {
```

Comment:
What's the rationale for putting this into the `Downloader` module?
As it is uses for the analysis, would the `analyzer` be a better fit? (or alternatively some `common` / `util` like module?

Reply:
As the downloader is (so far) focused on downloading the source code of packages I would also prefer a different location, like a utility or a client module.

Reply:
I went for the downloader as this actually downloads (significantly sized) data, not just "small stuff" like POMs in the analyzer. Also the main purpose of this code is the download itself, and not to work with the downloaded data.

I was also considering `ort-utils` (not `common-utils`) as that module already contains `OkHttpClientHelper`, but actually I would rather go for moving `OkHttpClientHelper` to the `downloader` as well (if that does not create undesirable module dependencies; I have not checked that yet).

Reply:
@mnonnenmacher I saw your comment only after writing mine. Would `ort-utils` work for both of you?

Reply:
My problem with that is that it doesn't match the description of the downloader and therefore having the feature and the discoclient dependency in here feels wrong:
> Fetches all source code of the projects and their dependencies, abstracting which Version Control System (VCS) or other means are used to retrieve the source code.

Reply:
> Would ort-utils work for both of you?

Yes, to me it's a better fit than the downloader.